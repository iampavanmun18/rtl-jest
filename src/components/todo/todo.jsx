export const Todo = ({ todo }) => {
  const { id, title, completed } = todo;
  const h1 = <h1>{title}</h1>;
  const text = completed ? <strike>{h1}</strike> : h1;
  return (
      <div data-tetsid={`todo-${id}`}>{text}</div>
  );
};
