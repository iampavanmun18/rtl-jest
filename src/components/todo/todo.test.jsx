import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Todo } from "./todo";

describe("test todo", () => {
  test("should render non-completed Todo", () => {
    const todo = { id: 1, title: "wash dishes", completed: false };
    render(<Todo todo={todo} />);
    // const textElement = screen.getByText("Hello from Todo!");
    const todoElement = screen.getByTestId("todo-1");
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("wash dishes");
    expect(todoElement).not.toContainHTML("<strike>");
  });

  test("should render completed Todo", () => {
    const todo = { id: 2, title: "wash car", completed: true };
    render(<Todo todo={todo} />);
    // const textElement = screen.getByText("Hello from Todo!");
    const todoElement = screen.getByTestId("todo-2");
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("wash car");
  });
});

describe("should render snapshot using react-test-renderer create method", () => {
  test("matches snapshot", () => {
    const todo = { id: 1, title: "wash dishes", completed: false };
    const tree = renderer.create(<Todo todo={todo} />).toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
  });
});
