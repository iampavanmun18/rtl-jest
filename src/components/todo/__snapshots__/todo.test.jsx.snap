// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render snapshot using react-test-renderer create method matches snapshot 1`] = `
<div
  data-tetsid="todo-1"
>
  <h1>
    wash dishes
  </h1>
</div>
`;
